app/src/main/java/edu/cascadia/mobile/apps/movies/MainActivity.java

```diff
@@ -53,10 +61,18 @@ public void onClick(View view) {
 
     }
 
+    // this was my issue , instance of the view model was never being reached so it remained null
+    // added this code to the oncreate.
+   //private void initViewModel() {
+        //mViewModel = ViewModelProviders.of(this)
+                 //.get(MainViewModel.class);
+    //}
+
     private void initRecyclerView(){
         mRecyclerView.setHasFixedSize(true);
         mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
-        mRecyclerView.setAdapter(new MoviesAdapter(mDatabase.movieDao().getMovies(),this));
+        mMoviesAdapter = new MoviesAdapter(movieData, this);
```

Comment:
Create and set the adapter in the ViewModel Observable's OnChange method

private void initViewModel() {
        final Observer<List<MovieEntity>> movieObserver =
                new Observer<List<MovieEntity>>() {
                    @Override
                    public void onChanged(@Nullable List<MovieEntity> movieEntities) {
                        mMovieList.clear();
                        mMovieList.addAll(movieEntities);

                        if (mAdapter == null) {
                            mAdapter = new MoviesAdapter(mMovieList,
                                    MainActivity.this);
                            mRecyclerView.setAdapter(mAdapter);
                        } else {
                            mAdapter.notifyDataSetChanged();
                        }

                    }
                };
        mViewModel.mMovies.observe(this, movieObserver);
    }